src/DSE.Open.Results/ResultBuilder.cs

```diff
@@ -9,12 +9,19 @@ namespace DSE.Open.Results;
 public abstract class ResultBuilder<TResult>
     where TResult : Result
 {
-    public ICollection<Notification> Notifications { get; } = new List<Notification>();
+    public virtual ResultStatus Status { get; set; }
+
+    public ICollection<Notification> Notifications { get; } = [];
 
     public virtual void MergeNotifications(Result result)
     {
         ArgumentNullException.ThrowIfNull(result);
 
+        if (Status == ResultStatus.Unspecified && result.Status != ResultStatus.Unspecified)
+        {
+            Status = result.Status;
+        }
+
```

Comment:
I don't like having ResultStatus being merged in a method called "MergeNotifications".

Reply:
Yes, if we do not want to handle by default, we could provide `MergeStatus` and `MergeNotificationsAndStatus` methods.